backend/handlers/handlers.go

```diff
@@ -954,6 +954,33 @@ func (handlers *Handlers) getAccountSummary(_ *http.Request) (interface{}, error
 
 		addChartData(account.Coin().Code(), timeseriesDaily, chartEntriesDaily)
 		addChartData(account.Coin().Code(), timeseriesHourly, chartEntriesHourly)
+
+		// HACK: We still use the latest prices from CryptoCompare for the account fiat balances
+		// above (displayed in the summary table). Those might deviate from the latest latest
+		// historical prices from coingecko, which results in different total balances in the chart
+		// and the summary table.
+		//
+		// As a temporary workaround, until we use only one source for all prices, we manually add
+		// one final datapoint reflecting the latest price. This can be removed once we stop using
+		// CryptoCompare.
+		now := time.Now().Unix()
+		price, err := handlers.backend.RatesUpdater().LastForPair(string(account.Coin().Code()), fiat)
```

Comment:
`LastForPair` has no timestamp in returned value, but it doesn't imply the data is recent enough.
Even if we switched to CoinGecko, there's still no guarantee what `LastForPair` would've returned is recent enough. It could be a data point from 1 hour ago but this code here wouldn't notice it.

Just saying, historical chart doesn't need to or maybe even can't always go up to `now`, regardless of the external API provider, cryptocompare or coingecko.

Reply:
That is very true. With coingecko at least we have a timestamp.